llvm/lib/Target/RISCV/RISCVFeatures.td

```diff
@@ -764,6 +771,62 @@ def FeatureStdExtSmepmp
     : SubtargetFeature<"smepmp", "HasStdExtSmepmp", "true",
                        "'Smepmp' (Enhanced Physical Memory Protection)", []>;
 
+def FeatureStdExtSsccptr
+    : SubtargetFeature<"ssccptr", "HasStdExtSsccptr", "true",
+                       "'Ssccptr' (Main memory supports page table reads)", []>;
+
+def FeatureStdExtShcounterenvw
+    : SubtargetFeature<"shcounterenw", "HasStdExtShcounterenw", "true",
+                       "'Shcounterenw' (Support writeable enables for any supproted counter)", []>;
+def FeatureStdExtSscounterenvw
+    : SubtargetFeature<"sscounterenw", "HasStdExtSscounterenw", "true",
+                       "'Sscounterenw' (Support writeable enables for any supproted counter)", []>;
```

Comment:
This description looks wrong.  Document says "Sscounterenw For any hpmcounter that is not read-only zero, the corresponding bit in scounteren must be writable."

Maybe:
"Support writeable enables for any supproted counter (with exception of zero registers)"

Also, typo: "supproted" should be "supported"

Reply:
Thanks. I had taken the shorter text from the very end of https://github.com/riscv/riscv-profiles/blob/main/rva23-profile.adoc

Reply:
Maybe "supported counter" meant it wasn't "read-only zero"?

Reply:
The difference between the two appears to be which counter must be writeable.

So maybe something like, "hcounter bits writeable for all non-zero hpmcounter bits" vs "scounter bits writeable for all non-zero hpmcounter bits"?

This is the case where having an actual spec for this would be really useful.

Reply:
This bit is about the counter enable not the counter itself.

The privilege spec says "All counters should be implemented, but a legal implementation is to make both the counter and its corresponding event selector be read-only 0."

And in the section in scounteren it says "scounteren must be implemented. However, any of the bits may be read-only zero, indicating reads to the corresponding counter will cause an exception when executing in U-mode."

I think this means without sscounterenw, it is legal to have a hpmcounter that is not read-only zero that is only exposable to S-mode because the scounteren bit to expose it to U-mode isn't writable.

Reply:
I honestly neither understand this, nor particular care.  I'm okay with any reasonable wording you want to use here - up to and including the previous identical wording for the two with the minor typo fix.  If we find better wording later, we can just update it then.